src/main/java/pl/betoncraft/betonquest/conditions/PartialDateCondition.java

```diff
@@ -0,0 +1,136 @@
+        String monthString = instruction.getOptional("month");
+        if (monthString != null) {
+            if (monthString.matches("[^-,]*")) {
+                int i = instruction.getInt(monthString, -1);
+                if (i < 1 || i > 12)
+                    throw new InstructionParseException(monthString + " is not a valid month");
+                month.add(i);
+            } else {
+                if (!monthString.matches("[0-9]{1,2}(-[0-9]{1,2})?(,[0-9]{1,2}(-[0-9]{1,2})?)*"))
+                    throw new InstructionParseException("could not parse months from " + monthString);
+                String[] args1 = monthString.split(",");
+                for (String arg : args1) {
+                    if (arg.contains("-")) {
+                        int index = arg.indexOf("-");
+                        int beginning = instruction.getInt(arg.substring(0,index), -1);
+                        int end = instruction.getInt(arg.substring(index + 1), -1);
+                        if (beginning >= end) throw new InstructionParseException("month " + beginning + " is before " + end);
+                        for (int i = beginning; i <= end; i++) {
+                            if (i < 1 || i > 12 ) throw new InstructionParseException(monthString + " contains invalid months");
+                            month.add(i);
+                        }
+                    } else {
+                        int i = instruction.getInt(arg, -1);
+                        if (i < 1 || i > 12 ) throw new InstructionParseException(monthString + " contains invalid months");
+                        month.add(i);
+                    }
+                }
+            }
+        }
+        String yearString = instruction.getOptional("year");
+        if (yearString != null) {
+            if (yearString.matches("[^-,]*")) {
+                int i = instruction.getInt(yearString, -1);
+                if (i < 1)
+                    throw new InstructionParseException(yearString + " is not a valid year");
+                year.add(i);
+            } else {
+                if (!yearString.matches("[0-9]+(-[0-9]+)?(,[0-9]+(-[0-9]+)?)*"))
+                    throw new InstructionParseException("could not parse years from " + yearString);
+                String[] args1 = yearString.split(",");
+                for (String arg : args1) {
+                    if (arg.contains("-")) {
+                        int index = arg.indexOf("-");
+                        int beginning = instruction.getInt(arg.substring(0,index), -1);
+                        int end = instruction.getInt(arg.substring(index + 1), -1);
+                        if (beginning >= end) throw new InstructionParseException("year " + beginning + " is before " + end);
+                        for (int i = beginning; i <= end; i++) {
+                            if (i < 1) throw new InstructionParseException(yearString + " contains invalid years");
+                            year.add(i);
```

Comment:
User tries to be funny and types `1-2147483647`, server crashes.

Reply:
💀 
Actually havn't thought about that as I thoght people who are smart enogh to write BetonQuest-Quest are also smart enogh not to try such expreiments.

But you are totally right we also have to write the plugin for Dan (a german saying, the long form would be "**D**ümmster **a**nzunehmender **N**utzer" which means as much as much as "the dumbest user to be assumed" 😄)

Reply:
"as much as much"? :D